src/sets.jl

```diff
@@ -807,6 +807,123 @@ end
 
 dimension(set::Complements) = 2 * set.dimension
 
+"""
+    AllDifferent(dimension::Int)
+
+The set corresponding to an all-different constraint.
+
+All expressions of a vector-valued function are enforced to take distinct
+values in the solution: for all pairs of expressions, their values must
+differ. This constraint is only defined for integer-valued expressions.
```

Comment:
I would remove `This constraint is only defined ...`. 

Presumably solvers will reject malformed models. But there's nothing stopping me from writing 
```Julia
model = Model()
@variable(model, x[1:2], Int)
@constriant(model, [x[1] + 1.0, 2x[2] - 2.0] in AllDifferent())
```